### DISCUSSION TITLE
Detecting a narrow beam of light on multiple sensors

### DISCUSSION THREAD

**Discussion by Ghiraa** [mitsuba-renderer/mitsuba3#1080] [Q&A]
Hi, I have a cylinder that is surrounded by sensors equally distributed among its edge pointing to the center. I am trying to simulate a "photon" by using a spotlight with a small cutoff angle and checking which has detected it (I avoid overlapping). For the purpose of this simulation, would you reckon a "batch" sensor would be more useful? I am currently defining a list of sensors and then rendering from each one, concatenating the resulting image into one.

**Comment by njroussel** [comment]
Hi @Ghiraa 

Yes, this is exactly what the `batch` sensor does. However, there is an upper bound to the amount of rays you can trace in a single kernel. You're obviously more likely to hit that limit if you're rendering multiple sensors at the same time.
